--- src/misc.c~	Wed Dec  5 11:59:51 2001
+++ src/misc.c	Mon Jul  8 21:19:12 2002
@@ -220,7 +220,7 @@
 void
 _gcry_log_printf (const char *fmt, ...)
 {
-  va_list arg_ptr = 0;
+  va_list arg_ptr ;
   
   if (fmt) 
     {
--- src/sexp.c~	Thu Jan 31 15:51:22 2002
+++ src/sexp.c	Mon Jul  8 21:24:10 2002
@@ -1111,7 +1111,7 @@
 gcry_sexp_sscan( GCRY_SEXP *retsexp, size_t *erroff,
 			    const char *buffer, size_t length )
 {
-    va_list dummy_arg_ptr = 0;
+    va_list dummy_arg_ptr;
 
     return sexp_sscan( retsexp, erroff, buffer, length, dummy_arg_ptr, 0 );
 }
