--- libgcrypt-org/mpi/config.links	2004-12-16 17:26:18.000000000 +0100
+++ libgcrypt-1.2.1/mpi/config.links	2005-02-01 23:34:38.000000000 +0100
@@ -130,20 +130,13 @@
 	path="pa7100 hppa1.1 hppa"
 	mpi_extra_modules="udiv-qrnnd"
 	;;
-    sparc64-*-linux-gnu)
-	# An extra rule because we have an report for this one only.
-	# Should be compared against the next GMP version
-	echo '/* configured for sparc64-*-linux-gnu */' >>./mpi/asm-syntax.h
-	path="sparc32v8 sparc32"
-	mpi_extra_modules="udiv"
-	;;
     sparc64-sun-solaris2*)
 	# Got a report that udiv is missing, so we try this one
 	echo '/* configured for sparc64-sun-solaris2 */' >>./mpi/asm-syntax.h
 	path="sparc32v8 sparc32"
 	mpi_extra_modules="udiv"
         ;;
-    sparc64-*-netbsd* | sparc64-*-freebsd* | sparc64-*-openbsd*)
+    sparc64-*-linux-gnu | sparc64-*-netbsd* | sparc64-*-freebsd* | sparc64-*-openbsd*)
        # There are no sparc64 assembler modules that work on the
        # *BSDs, so use the generic C functions.
        echo '/* No working assembler modules available */' >>./mpi/asm-syntax.h
