--- libgcrypt-1.1.90/src/libgcrypt.m4.orig	2003-11-11 14:05:52.000000000 +0100
+++ libgcrypt-1.1.90/src/libgcrypt.m4	2003-12-18 20:54:27.363009872 +0100
@@ -14,7 +14,7 @@
 dnl                   [ACTION-IF-FOUND [, ACTION-IF-NOT-FOUND ]]])
 dnl Test for liblibgcrypt and define LIBGCRYPT_CFLAGS and LIBGCRYPT_LIBS
 dnl
-AC_DEFUN(AM_PATH_LIBGCRYPT,
+AC_DEFUN([AM_PATH_LIBGCRYPT],
 [ AC_ARG_WITH(libgcrypt-prefix,
             AC_HELP_STRING([--with-libgcrypt-prefix=PFX],
                            [prefix where LIBGCRYPT is installed (optional)]),
