--- libgcrypt-1.6.0/random/rndlinux.c.orig	2013-12-16 11:44:25.000000000 +0100
+++ libgcrypt-1.6.0/random/rndlinux.c	2013-12-22 08:41:51.366559962 +0100
@@ -32,6 +32,7 @@
 #include <string.h>
 #include <unistd.h>
 #include <fcntl.h>
+#include <poll.h>
 #include "types.h"
 #include "g10lib.h"
 #include "rand-internal.h"
@@ -179,8 +180,7 @@
                  return with something we will actually use 100ms. */
   while (length)
     {
-      fd_set rfds;
-      struct timeval tv;
+      struct pollfd fds;
       int rc;
 
       /* If we collected some bytes update the progress indicator.  We
@@ -195,20 +195,10 @@
           any_need_entropy = 1;
         }
 
-      /* If the system has no limit on the number of file descriptors
-         and we encounter an fd which is larger than the fd_set size,
-         we don't use the select at all.  The select code is only used
-         to emit progress messages.  A better solution would be to
-         fall back to poll() if available.  */
-#ifdef FD_SETSIZE
-      if (fd < FD_SETSIZE)
-#endif
-        {
-          FD_ZERO(&rfds);
-          FD_SET(fd, &rfds);
-          tv.tv_sec = delay;
-          tv.tv_usec = delay? 0 : 100000;
-          if ( !(rc=select(fd+1, &rfds, NULL, NULL, &tv)) )
+        {
+          fds.fd = fd;
+          fds.events = POLLIN;
+          if ( !(rc=poll(&fds, 1, delay ? (delay*1000) : 100)) )
             {
               any_need_entropy = 1;
               delay = 3; /* Use 3 seconds henceforth.  */
@@ -216,7 +206,7 @@
             }
           else if( rc == -1 )
             {
-              log_error ("select() error: %s\n", strerror(errno));
+              log_error ("poll() error: %s\n", strerror(errno));
               if (!delay)
                 delay = 1; /* Use 1 second if we encounter an error before
                               we have ever blocked.  */
