--- libgcrypt-1.11.0/src/libgcrypt-config.in.orig	2024-06-23 13:19:55.926444925 +0200
+++ libgcrypt-1.11.0/src/libgcrypt-config.in	2024-06-23 17:10:04.991634797 +0200
@@ -165,11 +165,6 @@ if test "$echo_libs" = "yes"; then
     libdirs=""
     libs_final="$libs"
 
-    # Set up `libdirs'.
-    if test "x$libdir" != "x/usr/lib" -a "x$libdir" != "x/lib"; then
-	libdirs="-L$libdir"
-    fi
-
     # Set up `libs_final'.
     libs_final="$libs_final $gpg_error_libs"
 
