--- libgcrypt-1.10.1/src/libgcrypt-config.in.orig	2018-10-24 09:45:45.000000000 +0200
+++ libgcrypt-1.10.1/src/libgcrypt-config.in	2023-01-19 11:34:29.348466297 +0100
@@ -154,7 +154,7 @@
 
     tmp=""
     for i in $includes $cflags_final; do
-       if echo "$tmp" | fgrep -v -- "$i" >/dev/null; then
+       if echo "$tmp" | grep -F -v -- "$i" >/dev/null; then
            tmp="$tmp $i"
        fi
     done
@@ -165,17 +165,12 @@
     libdirs=""
     libs_final="$libs"
 
-    # Set up `libdirs'.
-    if test "x$libdir" != "x/usr/lib" -a "x$libdir" != "x/lib"; then
-	libdirs="-L$libdir"
-    fi
-
     # Set up `libs_final'.
     libs_final="$libs_final $gpg_error_libs"
 
     tmp=""
     for i in $libdirs $libs_final; do
-       if echo "$tmp" | fgrep -v -- "$i" >/dev/null; then
+       if echo "$tmp" | grep -F -v -- "$i" >/dev/null; then
            tmp="$tmp $i"
        fi
     done
